executor/internal/pdhelper/pd.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pdhelper
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/jellydator/ttlcache/v3"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/store/helper"
+	"github.com/pingcap/tidb/util"
+	"github.com/pingcap/tidb/util/sqlexec"
+)
+
+// GlobalPDHelper is the global variable for PDHelper.
+var GlobalPDHelper *PDHelper
+var globalPDHelperOnce sync.Once
+
+func init() {
+	GlobalPDHelper = newPDHelper()
+}
+
+// PDHelper is used to get some information from PD.
+type PDHelper struct {
+	wg                                       util.WaitGroupWrapper
+	cacheForApproximateTableCountFromStorage *ttlcache.Cache[string, float64]
+}
+
+func newPDHelper() *PDHelper {
+	cache := ttlcache.New[string, float64](
+		ttlcache.WithTTL[string, float64](30*time.Second),
+		ttlcache.WithCapacity[string, float64](1024*1024),
+	)
+	return &PDHelper{
+		cacheForApproximateTableCountFromStorage: cache,
+	}
+}
+
+// Start starts the PDHelper.
+func (p *PDHelper) Start() {
+	globalPDHelperOnce.Do(func() {
+		p.wg.Run(p.cacheForApproximateTableCountFromStorage.Start)
```

Comment:
Is it possible to run Start in init()?

Reply:
```init()``` is a bad code style.

refer https://github.com/uber-go/guide/blob/master/style.md#avoid-init

Reply:
it can be removed to ```Start()```.

Reply:
Looks like many "func init()" need to be revisited.